docs/propose_changelog.py

```diff
@@ -1,177 +1,119 @@
-            refactoring["plugin"].append(i.title + " #" + str(pr_nr))
-        elif "component: tools" in pr_labels_names:
-            refactoring["tools"].append(i.title + " #" + str(pr_nr))
-        else:
-            refactoring["other"].append(i.title + " #" + str(pr_nr))
-        continue
-    # all others are features
-    if "component: core" in pr_labels_names:
-        features["core"].append(i.title + " #" + str(pr_nr))
-        continue
-    elif "component: PMacc" in pr_labels_names:
-        features["pmacc"].append(i.title + " #" + str(pr_nr))
-        continue
-    elif "component: plugin" in pr_labels_names:
-        features["plugin"].append(i.title + " #" + str(pr_nr))
-        continue
-    elif "component: tools" in pr_labels_names:
-        features["tools"].append(i.title + " #" + str(pr_nr))
-        continue
-    # all leftovers are miscellaneous changes
-    if "documentation" in pr_labels_names:
-        misc["docs"].append(i.title + " #" + str(pr_nr))
-        continue
-    misc["other"].append(i.title + " #" + str(pr_nr))
-
-print("")
-print("**User Input Changes:**")
-for p in user["input"]:
-    print(" - " + p)
-
-print("")
-print("**New Features:**")
-print(" - PIC:")
-for p in features["core"]:
-    print("   - " + p)
-print(" - PMacc:")
-for p in features["pmacc"]:
-    print("   - " + p)
-print(" - plugins:")
-for p in features["plugin"]:
-    print("   - " + p)
-print(" - tools:")
-for p in features["tools"]:
-    print("   - " + p)
-for p in features["other"]:
-    print(" - " + p)
-
-print("")
-print("**Bug Fixes:**")
-print(" - PIC:")
-for p in bugs["core"]:
-    print("   - " + p)
-print(" - PMacc:")
-for p in bugs["pmacc"]:
-    print("   - " + p)
-print(" - plugins:")
-for p in bugs["plugin"]:
-    print("   - " + p)
-print(" - tools:")
-for p in bugs["tools"]:
-    print("   - " + p)
-for p in bugs["other"]:
-    print(" - " + p)
-
-print("")
-print("**Misc:**")
-print(" - refactoring:")
-print("   - PIC:")
-for p in refactoring["core"]:
-    print("     - " + p)
-print("   - PMacc:")
-for p in refactoring["pmacc"]:
-    print("     - " + p)
-print("   - plugins:")
-for p in refactoring["plugin"]:
-    print("     - " + p)
-print("   - tools:")
-for p in refactoring["tools"]:
-    print("     - " + p)
-for p in refactoring["other"]:
-    print("   - " + p)
-print(" - documentation:")
-for p in misc["docs"]:
-    print("   - " + p)
-for p in misc["other"]:
-    print(" - " + p)
+import sys
+import yaml
+
+
+def make_lambda(main_condition, component=None):
+    """Factory for lambdas encoding the categorisation conditions. Only needed to capture the current iteration by value."""
+    if component:
+        return lambda pr: contains_label(pr, f"component: {component}") and main_condition(pr)
+
+    # Okay, admittedly this is just a lazy way to sneak the 'other' category into this interface:
+    # A careful review would probably request to split this function up and give both of them better names.
```

Comment:
🤣

Reply:
Where is my [bottle of wine](https://www.bbc.com/news/articles/c84z2jqpvpko)?

Reply:
That is the third possible reaction to this, which I didn't see coming:
1. Don't read it and don't request changes.
2. Read it and request changes.
4. Read it but request something completely different.